Dockerfile

```diff
@@ -1,16 +1,87 @@
-FROM maven:3.5-jdk-8 as builder
+FROM ubuntu:18.04 as java
+
+# Install Java
+ARG JAVA_VERSION=8u171-1
+RUN \
+  apt-get update && \
+  apt-get install -y software-properties-common && \
+  echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections && \
+  add-apt-repository -y ppa:webupd8team/java && \
+  apt-get update && \
+  apt-get install -y oracle-java8-installer=${JAVA_VERSION}~webupd8~0 && \
+  rm -rf /var/lib/apt/lists/* && \
+  rm -rf /var/cache/oracle-jdk8-installer
+
+# Define commonly used JAVA_HOME variable
+ENV JAVA_HOME /usr/lib/jvm/java-8-oracle
+
+# install maven on top of java stage
+FROM java as build
+ARG MAVEN_VERSION=3.5.3
+ARG USER_HOME_DIR="/root"
+ARG SHA=b52956373fab1dd4277926507ab189fb797b3bc51a2a267a193c931fffad8408
+ARG BASE_URL=https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        curl \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
+  && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
+  && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha256sum -c - \
+  && tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1 \
+  && rm -f /tmp/apache-maven.tar.gz \
+  && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
+
+ENV MAVEN_HOME /usr/share/maven
+ENV MAVEN_CONFIG "$USER_HOME_DIR/.m2"
+
+COPY docker/mvn-entrypoint.sh /usr/local/bin/mvn-entrypoint.sh
+COPY docker/settings-docker.xml /usr/share/maven/ref/
+
+VOLUME "$USER_HOME_DIR/.m2"
+
+ENTRYPOINT ["/usr/local/bin/mvn-entrypoint.sh"]
+CMD ["mvn"]
+
+# install build dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        git \
+    && rm -rf /var/lib/apt/lists/*
+
 WORKDIR /iri
+
 COPY . /iri
 RUN mvn clean package
 
-FROM openjdk:jre-slim
-WORKDIR /iri
-COPY --from=builder /iri/target/iri-1.4.2.4.jar iri.jar
-COPY logback.xml /iri
-VOLUME /iri
+# execution image
+FROM java
```

Comment:
This is not an issue: the image referenced in this line is actually a build stage and build stages cannot have tags. I will make the build stage name for specific so that it is clear it's not a remote image, but instead a local one. Codacy will likely complain again.